librz/core/cmd/cmd_analysis.c

```diff
@@ -8812,3 +8443,220 @@ RZ_IPI RzCmdStatus rz_analysis_class_vtable_lookup_handler(RzCore *core, int arg
+	RzAnalysisBlock *bb = rz_analysis_find_most_relevant_block_in(core->analysis, core->offset);
+	if (!bb) {
+		RZ_LOG_ERROR("No basic block at 0x%" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_analysis_bb_info_print(core, bb, core->offset, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_basic_block_list_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	PJ *pj = state->mode == RZ_OUTPUT_MODE_JSON ? state->d.pj : NULL;
+	RzTable *table = state->mode == RZ_OUTPUT_MODE_TABLE ? state->d.t : NULL;
+
+	if (pj) {
+		pj_a(pj);
+	} else if (table) {
+		RzTableColumnType *s = rz_table_type("string");
+		RzTableColumnType *n = rz_table_type("number");
+		rz_table_add_column(table, n, "addr", 0);
+		rz_table_add_column(table, n, "size", 0);
+		rz_table_add_column(table, n, "traced", 0);
+		rz_table_add_column(table, n, "ninstr", 0);
+		rz_table_add_column(table, s, "jump", 0);
+		rz_table_add_column(table, s, "fail", 0);
+		rz_table_add_column(table, s, "fcns", 0);
+		rz_table_add_column(table, s, "calls", 0);
+		rz_table_add_column(table, s, "xrefs", 0);
+	}
+
+	RBIter iter;
+	RzAnalysisBlock *block;
+	rz_rbtree_foreach (core->analysis->bb_tree, iter, block, RzAnalysisBlock, _rb) {
+		RzList *xrefs = get_xrefs(block);
+		RzList *calls = get_calls(block);
+		switch (state->mode) {
+		case RZ_OUTPUT_MODE_JSON:
+			pj_o(pj);
+			pj_kn(pj, "addr", block->addr);
+			pj_kb(pj, "traced", block->traced);
+			pj_kn(pj, "ninstr", block->ninstr);
+			pj_kn(pj, "size", block->size);
+			if (block->jump != UT64_MAX) {
+				pj_kn(pj, "jump", block->jump);
+			}
+			if (block->fail != UT64_MAX) {
+				pj_kn(pj, "fail", block->fail);
+			}
+			if (xrefs) {
+				pj_ka(pj, "xrefs");
+				RzListIter *iter2;
+				ut64 *addr;
+				rz_list_foreach (xrefs, iter2, addr) {
+					pj_n(pj, *addr);
+				}
+				pj_end(pj);
+			}
+			if (calls) {
+				pj_ka(pj, "calls");
+				RzListIter *iter2;
+				ut64 *addr;
+				rz_list_foreach (calls, iter2, addr) {
+					pj_n(pj, *addr);
+				}
+				pj_end(pj);
+			}
+			pj_ka(pj, "fcns");
+			RzListIter *iter2;
+			RzAnalysisFunction *fcn;
+			rz_list_foreach (block->fcns, iter2, fcn) {
+				pj_n(pj, fcn->addr);
+			}
+			pj_end(pj);
+			pj_end(pj);
+			break;
+		case RZ_OUTPUT_MODE_TABLE: {
+			char *jump = block->jump != UT64_MAX ? rz_str_newf("0x%08" PFMT64x, block->jump) : strdup("");
+			char *fail = block->fail != UT64_MAX ? rz_str_newf("0x%08" PFMT64x, block->fail) : strdup("");
+			char *call = ut64join(calls);
+			char *xref = ut64join(calls);
+			char *fcns = fcnjoin(block->fcns);
+			rz_table_add_rowf(table, "xnddsssss",
+				block->addr,
+				block->size,
+				block->traced,
+				block->ninstr,
+				jump,
+				fail,
+				fcns,
+				call,
+				xref);
+			free(jump);
+			free(fail);
+			free(call);
+			free(xref);
+			free(fcns);
+		} break;
+		case RZ_OUTPUT_MODE_QUIET:
+			rz_cons_printf("0x%08" PFMT64x "\n", block->addr);
+			break;
+		default:
```

Comment:
I'd put here case STANDARD and then the default case just a rz_warn_if_reached so that if ever other modes are added we get a warning

Reply:
I've done this in many handlers